### ISSUE TITLE
question in batchsize

### ISSUE THREAD

**Issue by menghaoma** [AlanChou/Deep-Co-Training-for-Semi-Supervised-Image-Recognition#2]
Hi,

I am reading your code, which is a great implementation.
But I have a question. In the following code, why the batch_size in testloader is set to 100 but 1 in trainloader. 

testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size, shuffle=True, num_workers=2)
trainloader = torch.utils.data.DataLoader(trainset, batch_size=1, shuffle=False, num_workers=2)

**Comment by AlanChou** [comment]
Hi.

I knew it is going to confuse people. Sorry for the confusing coding style. 
The batch size you saw are not used for training. The batch size used for training are "S_batch_size" and "U_batch_size" (one for labelled data and one for unlabelled data.)

It was set to 1 because I was trying to get the training data one by one and separate them based on their true labels so that I can get a 10 dimension list where each dimension contains the indexes of the corresponded class.  It could be written in a more efficient way but this was the easiest way that came up to my mind.

To create "Labelled" and "Unlablled" data, one very important thing is that they have to be class-balanced. This is where the 10 dimension list can be used.
